app/validators/blacklist_entry_validator.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class BlacklistEntryValidator < ActiveModel::Validator
+  def validate(record)
+    return if (record.ip_literal.present? && record.ip_literal != "") ||
```

Comment:
The method `#present?` checks for a `nil` as well as a non-empty string. You don't need to do this. Just call `#present?`.

Reply:
Oooh that's cool!